extensions/amp-social-share/0.2/social-share.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as CSS from './social-share.css';
+import * as Preact from '../../../src/preact';
+import {Keys} from '../../../src/utils/key-codes';
+import {SocialShareIcon} from '../../../third_party/optimized-svg-icons/social-share-svgs';
+import {openWindowDialog} from '../../../src/dom';
+import {parseQueryString} from '../../../src/url';
+import {startsWith} from '../../../src/string';
+import {useResourcesNotify} from '../../../src/preact/utils';
+
+const DEFAULT_WIDTH = 60;
+const DEFAULT_HEIGHT = 44;
+
+/**
+ * @param {!JsonObject} props
+ * @return {PreactDef.Renderable}
+ */
+export function SocialShare(props) {
+  /**
+   * Handle key presses on the element.
+   * @param {!Event} event
+   * @private
+   */
+  function handleKeyPress(event) {
+    const {key} = event;
+    if (key == Keys.SPACE || key == Keys.ENTER) {
+      event.preventDefault();
+      handleActivation();
+    }
+  }
+
+  /** @private */
+  function handleActivation() {
+    const href = props['href'];
+    const target = props['target'] || '_blank';
+    if (!href) {
+      throw new Error('Clicked before href is set.');
+    }
+    if (startsWith(href, 'navigator-share:')) {
+      if (navigator.share !== undefined) {
+        throw new Error('navigator.share disappeared.');
+      }
+      // navigator.share() fails 'gulp check-types' validation on Travis
+      navigator['share'](parseQueryString(href.substr(href.indexOf('?'))));
+    } else {
+      const windowFeatures = 'resizable,scrollbars,width=640,height=480';
+      openWindowDialog(window, href, target, windowFeatures);
+    }
+  }
+
+  const styleType = `SOCIAL_SHARE_${props['type'].toUpperCase()}`;
+  const baseStyle = CSS.BASE_STYLE;
+  const backgroundStyle = CSS[styleType];
```

Comment:
I don't think this is possible with Closure. Please confirm @jridgewell. But from what I understand modules are not referenceable via map access. On the other hand, you can just put all of the known types to the separate map: `CSS.TYPE_STYLES`.

Reply:
I have no idea. We'll have to `gulp dist` to see if it mangles the module's namespace.

Reply:
`gulp dist` was able to find the background colors with this syntax but I'm happy to try out the separate map as well.